plugins/svgtools/include/detray/plugins/svgtools/illustrator.hpp

```diff
@@ -0,0 +1,176 @@
+            ret.add_object(svg);
+        }
+        return ret;
+    }
+
+    /// @brief Converts a detray volume in the detector to an actsvg svg.
+    /// @param identification the id of the svg object.
+    /// @param context the geometry context.
+    /// @param detector the detector.
+    /// @param index the index of the volume in the detector.
+    /// @param view the display view.
+    /// @returns SVG of the detector's volume.
+    template <typename view_t>
+    inline auto draw_volume(
+        const std::string& identification,
+        const typename detector_t::geometry_context& context,
+        const size_t index, const view_t& view) const {
+        const auto volume =
+            _detector.volume_by_index(static_cast<detray::dindex>(index));
+        auto p_volume = svgtools::conversion::volume<point3_container>(
+            context, _detector, volume);
+        svgtools::styling::apply_style(p_volume, _style);
+        return actsvg::display::volume(identification, p_volume, view);
+    }
+
+    /// @brief Converts a collection of detray volumes in the detector to an
+    /// actsvg svg.
+    /// @param identification the id of the svg object.
+    /// @param context the geometry context.
+    /// @param detector the detector.
+    /// @param indices the collection of volume indices in the detector to
+    /// convert.
+    /// @param view the display view.
+    /// @returns SVG of the detector's volumes.
+    template <typename iterator_t, typename view_t>
+    inline auto draw_volumes(
+        const std::string& identification,
+        const typename detector_t::geometry_context& context,
+        const iterator_t& indices, const view_t& view) const {
+        actsvg::svg::object ret;
+        ret._tag = "g";
+        ret._id = identification;
+        for (const auto index : indices) {
+            const auto svg =
+                draw_volume(identification + "_volume" + std::to_string(index),
+                            context, index, view);
+            ret.add_object(svg);
+        }
+        return ret;
+    }
+
+    /// @brief Converts a detray detector to an actsvg svg.
+    /// @param identification the id of the svg object.
+    /// @param context the geometry context.
+    /// @param detector the detector.
+    /// @param view the display view.
+    /// @returns SVG of the detector.
+    template <typename view_t>
+    inline auto draw_detector(
+        const std::string& identification,
+        const typename detector_t::geometry_context& context,
+        const view_t& view) const {
+        actsvg::svg::object ret;
+        ret._tag = "g";
+        ret._id = identification;
+        for (std::size_t index = 0; index < _detector.volumes().size();
+             index++) {
+            const auto svg =
+                draw_volume(identification + "_volume" + std::to_string(index),
+                            context, index, view);
+            ret.add_object(svg);
+        }
```

Comment:
Just an idea, I don't know if it works and in the interest of getting this in we should probably leave it as it is :)
```suggestion
        auto indices = detray::views::iota(std::size_t{0u}, _detector.volumes().size());
        return draw_volumes(identification, context, indices, view);
```
Also, you would need to ```#include "detray/utils/ranges.hpp"```